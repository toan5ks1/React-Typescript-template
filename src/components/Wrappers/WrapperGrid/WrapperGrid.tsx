import { ReactElement } from "react";

import { Box } from "@mui/material";

const WrapperGrid = (props: { children: ReactElement }) => {
    return <Box style={{ overflow: "hidden" }}>{props.children}</Box>;
};

export default WrapperGrid;
