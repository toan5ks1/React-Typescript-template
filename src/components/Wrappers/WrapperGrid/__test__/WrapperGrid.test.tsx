import WrapperGrid from "../WrapperGrid";

import { render, RenderResult, screen } from "@testing-library/react";

describe("<WrapperGrid />", () => {
    let wrapper: RenderResult;

    const props = {
        children: <div>Test</div>,
    };

    beforeEach(() => {
        wrapper = render(<WrapperGrid {...props} />);
    });

    it("should exist children", () => {
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });
});
