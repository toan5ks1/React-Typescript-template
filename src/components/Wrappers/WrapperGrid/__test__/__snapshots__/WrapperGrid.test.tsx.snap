// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WrapperGrid /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root emotion-0"
    style="overflow: hidden;"
  >
    <div>
      Test
    </div>
  </div>
</div>
`;
